src/Native/System.Native/pal_errno.cpp

```diff
@@ -330,6 +340,16 @@ extern "C" int32_t SystemNative_ConvertErrorPalToPlatform(Error error)
             return ETXTBSY;
         case PAL_EXDEV:
             return EXDEV;
+        case PAL_EPFNOSUPPORT:
+            return EPFNOSUPPORT;
+        case PAL_ESOCKTNOSUPPORT:
+            return ESOCKTNOSUPPORT;
+        case PAL_ESHUTDOWN:
+            return ESHUTDOWN;
+        case PAL_EHOSTDOWN:
+            return EHOSTDOWN;
+        case PAL_ENODATA:
+            return ENODATA;
```

Comment:
@stephentoub, in #2864 `ENODATA` was removed. Should we add an `ifdef` check to omit this case for FreeBSD?
(this was reported at https://github.com/dotnet/coreclr/issues/6115#issuecomment-231114749)